briskV3/hal.c

```diff
@@ -12,120 +17,65 @@
-{
-	GPIO_InitTypeDef GPIO_InitStruct = {0};
-	GPIO_InitStruct.Pin = ROBUS_TX_DETECT_Pin;
-	GPIO_InitStruct.Mode = GPIO_MODE_IT_FALLING;
-	GPIO_InitStruct.Pull = GPIO_PULLUP;
-	HAL_GPIO_Init(ROBUS_TX_DETECT_Port, &GPIO_InitStruct);
-}
-/**
- * \fn HAL_DisableTxLockDetection(void)
- * \brief PTP interrupt management
- */
-void HAL_DisableTxLockDetection(void)
-{
-	GPIO_InitTypeDef GPIO_InitStruct = {0};
-	GPIO_InitStruct.Pin = ROBUS_TX_DETECT_Pin;
-	GPIO_InitStruct.Mode = GPIO_MODE_INPUT;
-	GPIO_InitStruct.Pull = GPIO_NOPULL;
-	HAL_GPIO_Init(ROBUS_TX_DETECT_Port, &GPIO_InitStruct);
-}
-/**
- * \fn HAL_is_tx_lock(void)
- * \brief PTP interrupt management
- */
-void HAL_LockTx(uint8_t lock)
+/////////////////////////Luos Library Needed function///////////////////////////
+/******************************************************************************
+ * @brief Luos HAL general initialisation
+ * @param None
+ * @return None
+ ******************************************************************************/
+void LuosHAL_Init(void)
 {
-	ctx.tx_lock = lock;
+	// Flash Initialisation
+	LuosHAL_FlashInit();
+
+	// CRC Initialisation
+	LuosHAL_CRCInit();
+
+    //Com Initialisation
+	LuosHAL_ComInit(DEFAULTBAUDRATE);
+
+    //Timout Initialisation
+	LuosHAL_TimeoutInit();
+
+    //IO Initialisation
+	LuosHAL_GPIOInit();
 }
-/**
- * \fn HAL_GPIO_EXTI_Callback(uint16_t GPIO_Pin)
- * \brief PTP interrupt management
- */
-void HAL_GPIO_EXTI_Callback(uint16_t GPIO_Pin)
+/******************************************************************************
+ * @brief Luos HAL general disable IRQ
+ * @param None
+ * @return None
+ ******************************************************************************/
+void LuosHAL_IrqStatus(uint8_t Enable)
 {
-    if (GPIO_Pin == ROBUS_PTPA_Pin)
-    {
-        ptp_handler(BRANCH_A);
-        return;
-    }
-    if (GPIO_Pin == ROBUS_PTPB_Pin)
-    {
-        ptp_handler(BRANCH_B);
-        return;
-    }
-	if (GPIO_Pin == ROBUS_TX_DETECT_Pin)
+	if(Enable == true)
+	{
+		__enable_irq();
+	}
+	else
 	{
-		HAL_LockTx(true);
-		return;
+		__disable_irq();
 	}
 }
-/**
- * \fn unsigned char crc(unsigned char* data, unsigned char size)
- * \brief generate a CRC
- *
- * \param *data data table
- * \param size data size
- *
- * \return CRC value
- */
-void set_baudrate(unsigned int baudrate)
+/******************************************************************************
+ * @brief Luos HAL general systick tick at 1ms
+ * @param None
+ * @return tick Counter
+ ******************************************************************************/
+uint32_t LuosHAL_GetSystick(void)
+{
+	return HAL_GetTick();
+}
+/******************************************************************************
+ * @brief Luos HAL Initialize Generale communication inter node
+ * @param Select a baudrate for the Com
+ * @return none
+ ******************************************************************************/
+void LuosHAL_ComInit(uint32_t baudrate)
```

Comment:
All sub-init function should be static and not declared on the .h. The only file able to call it is this one.

Reply:
all founction init is in static exept ComInit du to the fact that bauderate can be change by msg